namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftNamerInterface.scala

```diff
@@ -265,22 +259,14 @@ class ThriftNamerInterface(
     }
   }
 
-  private[this] val bindingCacheMu = new {}
-  private[this] var bindingCache: Map[(Ns, Dtab, Path), BindingObserver] = Map.empty
-  private[this] def observeBind(ns: String, dtab: Dtab, path: Path): BindingObserver =
-    bindingCacheMu.synchronized {
-      val key = (ns, dtab, path)
-      bindingCache.get(key) match {
-        case Some(obs) =>
-          Trace.recordBinary("namerd.srv/bind.cached", true)
-          obs
-        case None =>
-          Trace.recordBinary("namerd.srv/bind.cached", false)
-          val obs = BindingObserver(interpreters(ns).bind(dtab, path), stamper)
-          bindingCache += (key -> obs)
-          obs
-      }
-    }
+  private[this] def observeBind(ns: Ns, dtab: Dtab, path: Path) =
+    BindingObserver(interpreters(ns).bind(dtab, path), stamper)
+  private[this] val bindingCache = new ObserverCache[(String, Dtab, Path), NameTree[Name.Bound]](
+    activeCapacity = 100,
+    inactiveCapacity = 10,
+    stats = stats.scope("binding-cache"),
```

Comment:
resultant stats will be e.g. "binding-cache/obsever-cache/active-size" -- can we tighten this up a bit?  Do we even need the observer-cache scope anymore?  Also, I would omit "-size" -- key verbosity comes with a cost at scale.

Can this just be bindingcache/active? (seems more idiomatic with rest of finagle stats)